PHR-13265 - Fix select element component.

BalintViragJ opened the pull request:
[PHR-13265](https://pkbdev.atlassian.net/browse/PHR-13265)

[PHR-13265]: https://pkbdev.atlassian.net/browse/PHR-13265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

BalintViragJ committed:
Fix select element component.
PHR-13265

idasbiste commented on src/app/elements/element-selection-tab/select-element/select-element-dialog/select-element-dialog.component.ts:
```diff
@@ -11,19 +14,29 @@ import {ElementsComponent} from "../../../elements.component";
+  for (let i = 0; i < this.bufferArray.length; i++) {
+    this.selectOptionValues[i] = this.bufferArray[i]
+  }
+}
```
The function name is misleading. It says fill up the buffer array but in fact you are populating the options value.

BalintViragJ replied:
Changed it, thank you!

idasbiste commented on src/app/elements/element-selection-tab/select-element/select-element-dialog/select-element-dialog.component.ts:
```diff
@@ -11,19 +14,29 @@ import {ElementsComponent} from "../../../elements.component";
   }
 
   ngOnInit() {
+    for (let i = 0; i < this.selectOptionValues.length; i++) {
+      this.bufferArray.push(this.selectOptionValues[i])
+    }
```
Any reason this is done in `ngOnInit` instead of inside the constructor? Also, because the array contains a primitive type you could do
```typescript
this.bufferArray = [...this.selectOptionValues]
```

BalintViragJ replied:
> this.bufferArray = [...this.selectOptionValues]

⬆️ I've just learned something new, thank you! 
No specific reason for the `ngOnInit`, I just made a mistake. Relocated the functionality to the constructor. Thank you

BalintViragJ committed:
Fixes:
1. Remove buttons are now bind to a single select option so that specified elements can be removed instead of removing the last option from the list.
2. Made changes requested by reviewer
PHR-13265

BalintViragJ committed:
Fix radio and checkbox elements + fix element ordering on generated page.
PHR-13268

dorogimate commented:
Added this comment to the JIRA tickets as well, and highlighted the parts that needs to be addressed. Acceptance Criteria feedback after checking changes manually:
    1. Works, just one note here, when I add a new option, it is empty at first. And if I save it without any content, then VALUE is put there. I think that is okay like this.
    2. It works nicely. One idea here, **it would be nice to add another button called “Add after” next to each option**. E.g. if someone wants to add an option between 6th and 7th, then it is easier.
    3. Works as it should!
    4. Checked and working nicely.
    5. Still all the options have checked class attribute. **Discussed it and that can be actually removed, so no need to add a button that can set which option is the checked one**. Maybe the first can be by default. Also if all the option are checked, the first one will be checked (meaning it is displayed on the drop down box). So it would be cleaner if only the first one would have checked class name, but it works in this way as well, it is just not that clean.
    7. Nice to have - not compulsory

I think the after testing manually the changes are good. Only need to discuss the case with the checked class attribute (which actually working nicely, it is just the html that can be cleaned up a little bit)

BalintViragJ committed:
Remove "check" values from class attributes.
PHR-13268

BalintViragJ commented:
@dorogimate 

> Works, just one note here, when I add a new option, it is empty at first. And if I save it without any content, then VALUE is put there. I think that is okay like this.

I also think that it's fine like this. Users can easily swap the values after adding a new option to the end of the list.

>  It works nicely. One idea here, it would be nice to add another button called “Add after” next to each option. E.g. if someone wants to add an option between 6th and 7th, then it is easier.

I agree. Let's do it in a follow up after we get some feedback from real users. 

> So it would be cleaner if only the first one would have checked class name, but it works in this way as well, it is just not that clean

But what if it's not what the users want? I think we can keep it like this and then after getting feedback we can change this in a follow up.

dorogimate commented on src/app/elements/element-selection-tab/select-element/select-element-dialog/select-element-dialog.component.ts:
```diff
@@ -1,37 +1,47 @@
 
-  removeOption(){
+  removeOption(indexOfTheElement: number){
+    console.log(indexOfTheElement)
```
nb: console log can be removed, I can do that in another CP ticket.

dorogimate commented on src/app/elements/element-selection-tab/select-element/select-element.component.ts:
```diff
@@ -44,9 +41,10 @@ export class SelectElementComponent implements ElementType, OnInit {
     let options = '';
+    console.log('selectOptionValues[i]: ' + this.selectOptionValues)
     for (let i = 0; i < this.selectOptionValues.length; i++) {
-
-      options += '    <option class="cp-select-option checked" value="VALUE">' + this.selectOptionValues[i] + '\n' +
+      console.log('selectOptionValues[i]: ' + this.selectOptionValues[i])
```
nb: few left over console logs I think

dorogimate commented on src/app/elements/element-selection-tab/select-element/select-element.component.ts:
```diff
@@ -60,6 +58,7 @@ export class SelectElementComponent implements ElementType, OnInit {
     dialogRef.afterClosed().subscribe(result => {
       if (result[0] !== '') {
         this.label = result[0];
+        //this.selectOptionValues = result[1]
```
nb: left over comment

dorogimate reviewed (approved):
Few non blocking comments, I can do those in another CP builder ticket. Let me know if I should deal with them.

BalintViragJ committed:
Fix popup window
PHR-13263

dorogimate committed:
Merge pull request #5 from patientsknowbest/PHR-13263_fix_popup_window

PHR-13263 - Fix popup window

dorogimate committed:
Merge pull request #4 from patientsknowbest/PHR-13268_fix_radio_and_checkbox

PHR-13268  - Fix radio and checkbox

dorogimate commented:
I needed to re-review the PR, cause received an error, when tried to merge it. It was because other PRs were merged into this one and I needed to re-approve it in order to be able to merge.
